interactions/api/gateway/client.py

```diff
@@ -450,18 +447,15 @@ def __modify_guild_cache():
                             return
                         _obj = getattr(guild, f"{model_name.lower()}s", None)
                         if _obj is not None and isinstance(_obj, list):
-                            _data = getattr(obj, model_name, None)
-
                             if "_create" in name or "_add" in name:
                                 _obj.append(obj)
-
                             for index, __obj in enumerate(_obj):
                                 if __obj.id == id:
                                     if "_remove" in name or "_delete" in name:
                                         _obj.remove(__obj)
 
                                     elif "_update" in name and hasattr(obj, "id"):
-                                        _obj[index] = _data
+                                        _obj[index] = obj
```

Comment:
^Can you explain this changes?

Reply:
`_data` is None because you tries to get from channel instanse `Channel`.
Then in `guild.channels` instead of normal channel appears `None` and then you get error when edit channel

![image](https://user-images.githubusercontent.com/25762802/180659187-75945dfd-bdc0-43e5-b1c6-92b58815f7b3.png)

Reply:
I see